autoemulate/experimental/emulators/neural_processes/conditional_neural_process.py

```diff
@@ -447,8 +440,8 @@ def _predict(self, x: TensorLike) -> DistributionLike:
             raise ValueError(msg)
 
         # Unsqueeze the batch dimension for x_train, y_train and x_target
-        x_train = self.x_train.unsqueeze(0)
-        y_train = self.y_train.unsqueeze(0)
+        x_train = self.x_train.unsqueeze(0)  # type: ignore PGH003
+        y_train = self.y_train.unsqueeze(0)  # type: ignore PGH003
```

Comment:
Perhaps we can do:
```suggestion
        assert isinstance(self.x_train, TensorLike)
        assert isinstance(self.y_train, TensorLike)
        x_train = self.x_train.unsqueeze(0)
        y_train = self.y_train.unsqueeze(0)
```
since we have checked that it is fitted and x_train and y_train are assigned during the fit.